src/components/common/Viewer.jsx

```diff
@@ -263,57 +254,47 @@ export default class Viewer extends Component {
         BackendAPI.removeViewerSelectionsChange(this.viewerSelectionListener);
         BackendAPI.removeViewerTransformChange(this.viewerTransformListener);
         BackendAPI.removeCustomScaleChange(this.customScaleListener);
+
+        const activePage = decodeURI(this.props.location.pathname)
+            .split('/')
+            .slice(-1)[0];
         BackendAPI.removeActiveFeaturesChange(
-            this.state.activePage,
+            activePage,
             this.activeFeaturesListener
         );
         BackendAPI.removeSpriteSettingsChange(this.spriteSettingsListener);
         this.destroyGraphics();
     }
-    /*
-	shouldComponentUpdate(nextProps, nextState) {
-		return false;
-	}
-*/
-    onResize(width, height) {
-        let bbox = this.zoomSelection
+
+    onResize() {
+        const bbox = this.zoomSelection
             .select(function () {
                 return this.parentNode;
             })
             .node()
             .getBoundingClientRect();
-        console.log(this.props.name, 'onResize', width, height, bbox);
+
         let dw = bbox.width - this.w - VIEWER_MARGIN;
         let dh = bbox.height - this.h - VIEWER_MARGIN;
-        if (
-            (bbox.width != 0 &&
-                (2 * VIEWER_MARGIN < dw || dw < -2 * VIEWER_MARGIN)) ||
-            (bbox.height != 0 &&
-                (2 * VIEWER_MARGIN < dh || dh < -2 * VIEWER_MARGIN))
-        ) {
-            this.w = bbox.width - VIEWER_MARGIN;
-            this.h = bbox.height - VIEWER_MARGIN;
-            this.resizeContainer();
-        }
+
+        this.w = bbox.width - VIEWER_MARGIN;
+        this.h = bbox.height - VIEWER_MARGIN;
+        this.resizeContainer();
```

Comment:
This is great to be fixed, I noticed one weird behaviour though.

1) Select loom to show scatter viewer
2) Resize window from bottom

The browser seems to constantly update the size of the page 1 pixel at a time and the viewer displays as a black square during this.

Windows 10, Chrome Version 85.0.4183.121

Reply:
Would you believe me if I said this is being caused by `AppHeader`? That's what the profiler says (WAT!)

Reply:
Oops you would be right not to believe me, that's not it